packages/@blockprotocol/graph/src/type-resolver/entity-type-to-typescript/deduplicate-ts-strings.ts

```diff
@@ -0,0 +1,51 @@
+const deduplicateTypeScriptString = (
+  firstString: string,
+  secondString: string,
+) => {
+  let deduplicatedSecondString = secondString;
+
+  const textLines = secondString.split("\n");
+  for (let i = 0; i < textLines.length; i++) {
+    const line = textLines[i];
+    const isLineStartingType = line?.startsWith("export type");
+    const isLineStartingInterface = line?.startsWith("export interface");
+
+    if (line && (isLineStartingType || isLineStartingInterface)) {
+      const lineOpensBrace = isLineStartingInterface || line.includes(" = {"); // @todo very long type names might break this
```

Comment:
I think we might hit this rather quickly, we might want to try and solve this soon.

Reply:
Done in https://github.com/blockprotocol/blockprotocol/pull/861/commits#:~:text=handle%20very%20long%20type%20names%20when%20deduplicating